### ISSUE TITLE
In case of restarting Tendermint, Can we distinguish between old and new transaction

### ISSUE THREAD

**Issue by ajay-arya** [tendermint/tendermint#8040]
I am using Tendermint as a single node and I am new to Tendermint as well as Blockchain. 

While using Tedermint, whenever I restart the blockchain with previous transaction logs. It will redo all the transactions. 
I want to know if there is any way I can able to know that this transaction is taken or redoing from the log.

Thanks in advance
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by creachadair** [comment]
I do not believe there is any way to distinguish uncommitted transactions being replayed after a restart, from transactions that have just arrived without a restart. 

As a general rule, an application should not need to care about the distinction: If a transaction hasn't been committed yet, it's possible that transaction will _never_ be added to a block, and thus the application can't safely update its state based on that transaction.

An application could choose to record which transactions it's seen (via `CheckTx`) that have not yet been delivered in a block. It's not clear to me what problem that would solve, though: The point of consensus is to ensure that applications only process transactions that the whole network has (provably) agreed to. Transactions not yet committed are not yet agreed.

**Comment by ajay-arya** [comment]
Thank you, @creachadair. 

I knew it was not possible but the Tendermint is so fast that I am trying to use it in place of a regular server. It was just experimentation that I was doing, where I was inserting a few data into the database and tried to update the same data outside of the Tendermint.
